Hide defense for non tanks

dejoblue opened the pull request:
Hid the Defense category and its relevant stats for non tanks.

Changed AttackCategory to OffenseCategory to reduce confusion.

dejoblue committed:
Hide Defense for Non Tanks

Hid the Defense category and its relevant stats for non tanks.

Changed AttackCategory to OffenseCategory to reduce confusion.

Kakjens committed:
Armor changes place instead of becoming hidden for non-tanks

Wrote without thunking much about the most efficient implementation.
Block and parry still can appear for guardian druids. Seems like druids
can't parry even with Deflection Draught from Darkmoon Faire (Deftness
Draught which increases dodge works).

Kakjens commented:
Since even Blizzard shows armor for non-tanks, we also will.
Attributes with 2 stats seem lonely.

dejoblue committed:
Changed Default Non Tank Armor placement

Changed Default Non Tank Armor placement

dejoblue commented:
Changed Default Non Tank Armor placement to be a main attribute under Str Agi Int Sta.

Yeah, I didn't really want o make two tables for one stat, but this works for now.

Yeah, Druids can't parry or block.

Merge this unless you find anything other changes that need to be made and let's get this pushed as a release. I'll get the Russian up as well before we do. I'll probably end up doing it after I get the RU up.

dejoblue committed:
Fixed Relevant Stats Resetting order

Fixed Relevant Stats Resetting order
Remnants from my fix made the stat order reset when clicking relevant
stats button. Removing it fixes it.

Kakjens commented:
That was really careless of me to un-commented that line, and even copy-paste from DCS_Login_Initialization(). Sorry! Thinking of how to write [certain branch](https://github.com/dejoblue/DejaCharacterStats/tree/first-time) might have influenced my thought process.
Merging.
